lume_model/tests/pytorch/test_model.py

```diff
@@ -0,0 +1,332 @@
+    assert_variables_updated(
+        california_test_x_dict["HouseAge"].item(),
+        4.063651,
+        cal_model,
+        "HouseAge",
+        "MedHouseVal",
+    )
+
+
+@pytest.mark.parametrize(
+    "test_idx,expected",
+    [
+        (0, torch.tensor(4.063651, dtype=torch.double)),
+        (1, torch.tensor(2.7774928, dtype=torch.double)),
+        (2, torch.tensor(2.792812, dtype=torch.double)),
+    ],
+)
+def test_california_housing_model_execution_diff_values(
+    test_idx: int,
+    expected: torch.Tensor,
+    california_test_x: torch.Tensor,
+    cal_model: PyTorchModel,
+):
+    test_input = {
+        key: california_test_x[test_idx][idx]
+        for idx, key in enumerate(cal_model.features)
+    }
+
+    results = cal_model.evaluate(test_input)
+
+    assert torch.isclose(results["MedHouseVal"], expected)
+    assert_variables_updated(
+        test_input["HouseAge"].item(),
+        expected.item(),
+        cal_model,
+        "HouseAge",
+        "MedHouseVal",
+    )
+
+
+def test_california_housing_model_execution_no_transformation(
+    california_test_x_dict: Dict[str, torch.Tensor],
+    california_model_kwargs: Dict[str, Union[List, Dict, str]],
+):
+    # if we don't pass in an output transformer, we expect to get the untransformed
+    # result back
+    new_kwargs = deepcopy(california_model_kwargs)
+    new_kwargs["output_transformers"] = []
+    cal_model = PyTorchModel(**new_kwargs)
+
+    results = cal_model.evaluate(california_test_x_dict)
+
+    assert torch.isclose(
+        results["MedHouseVal"], torch.tensor(1.8523695, dtype=torch.double)
+    )
+    assert_variables_updated(
+        california_test_x_dict["HouseAge"].item(),
+        1.8523695,
+        cal_model,
+        "HouseAge",
+        "MedHouseVal",
+    )
+
+
+def test_california_housing_model_execution_missing_input(
+    caplog, california_test_x_dict: Dict[str, torch.Tensor], cal_model: PyTorchModel
+):
+    missing_dict = deepcopy(california_test_x_dict)
+    del missing_dict["Longitude"]
+
+    with caplog.at_level(logging.INFO):
+        results = cal_model.evaluate(missing_dict)
+        assert len(caplog.records) == 1
+        assert caplog.records[0].levelname == "INFO"
+        assert (
+            caplog.records[0].message
+            == "'Longitude' missing from input_dict, using default value"
+        )
+
+
+def test_differentiability(
+    california_test_x_dict: Dict[str, torch.Tensor], cal_model: PyTorchModel
+):
+    differentiable_dict = deepcopy(california_test_x_dict)
+    for value in differentiable_dict.values():
+        value.requires_grad = True
+
+    results = cal_model.evaluate(differentiable_dict)
+
+    # if we maintain differentiability, we should be able to call .backward()
+    # on a model output without it causing an error
+    for key, value in results.items():
+        try:
+            value.backward()
+            assert value.requires_grad
+        except AttributeError as exc:
```

Comment:
I don't think you need the try/except statement here, the test will fail normally if value.backward() raises an error